skimage/future/_thin_plate_splines.py

```diff
@@ -0,0 +1,323 @@
+import numpy as np
+import scipy as sp
+
+from .._shared.utils import check_nD
+
+
+class TpsTransform:
+    """Thin-plate spline transformation.
+
+    Given a set of control points (source and destination points), this class
+    can be used to estimate the thin-plate spline (TPS) transformation, which
+    transforms the source points into the destination points.
+    It interpolates a surface that passes through each control point.
+    Control points are seen as position constraints on a bending surface. The ideal
+    surface is the one that bends the least.
+
+    Attributes
+    ----------
+    spline_mappings : (N, D) array_like
+        Spline mappings for every control point. These describe a deformation.
+    src : (N, 2) array_like
+        Coordinates of control points in source image.
+
+    References
+    ----------
+    .. [1] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+           decomposition of deformations," IEEE Transactions on pattern analysis
+           and machine intelligence 11.6 (1989): 567–585.
+           DOI:`10.1109/34.24792`
+           https://user.engineering.uiowa.edu/~aip/papers/bookstein-89.pdf
+
+    Examples
+    --------
+    >>> import skimage as ski
+
+    Define source and destination control points:
+
+    >>> src = np.array([[0, 0], [0, 5], [5, 5],[5, 0]])
+    >>> dst = np.roll(src, 1, axis=0)
```

Comment:
`roll` is lesser known, and it may not be obvious to the reader what this means. At minimum, print `dst`.

Reply:
Also, is this a good example? Looks like a 180 degree flip, so does not really require a thin plate spline.

Reply:
It's good in the sense that it's easy to verify it's correctness. What would a good example constitute in your eyes? And agreed that `np.roll` might be lesser known? Though, I'd think that `roll(..., axis=0)` is pretty clear? If we keep it, we could use
```suggestion
    >>> dst = np.roll(src, shift=1, axis=0)
```
to make it clearer.